Code/GraphMol/FileParsers/CMLWriter.cpp

```diff
@@ -0,0 +1,298 @@
+    atom.put("<xmlattr>.id", boost::format{"%1%%2%"} % atom_id_prefix % i);
+    if (a->getAtomicNum()) {
+      atom.put("<xmlattr>.elementType", a->getSymbol());
+    } else {
+      atom.put("<xmlattr>.elementType", "Du");  // dummy
+    }
+
+    const auto charge = a->getFormalCharge();
+    mol_formal_charge += charge;
+    atom.put("<xmlattr>.formalCharge", charge);
+
+    atom.put("<xmlattr>.hydrogenCount", a->getTotalNumHs());
+
+    const auto isotope = a->getIsotope();
+    if (isotope) {
+      atom.put("<xmlattr>.isotopeNumber", isotope);
+    }
+
+    const auto n_rad_es = a->getNumRadicalElectrons();
+    mol_num_radical_electrons += n_rad_es;
+
+    if (!mol.getNumConformers()) {
+      continue;
+    }
+
+    const auto& conf = rwmol.getConformer(confId);
+    const auto& pos = conf.getAtomPos(i);
+    boost::format xyz_fmt{"%.6f"};
+
+    if (!conf.is3D()) {
+      atom.put("<xmlattr>.x2", xyz_fmt % pos.x);
+      atom.put("<xmlattr>.y2", xyz_fmt % pos.y);
+      continue;
+    }
+
+    atom.put("<xmlattr>.x3", xyz_fmt % pos.x);
+    atom.put("<xmlattr>.y3", xyz_fmt % pos.y);
+    atom.put("<xmlattr>.z3", xyz_fmt % pos.z);
+
+    // atom/@atomParity if chiral
+    // http://www.xml-cml.org/convention/molecular#atom-atomParity
+    const auto chiral = a->getChiralTag();
+    if (chiral == Atom::CHI_UNSPECIFIED || chiral == Atom::CHI_OTHER) {
+      continue;
+    }
+
+    std::vector<RDGeom::Point3D> xyzs;
+    std::vector<unsigned> neighbors;
+    for (auto nbri : boost::make_iterator_range(mol.getAtomBonds(a))) {
+      const auto* const b = mol[nbri];
+      const auto o = b->getOtherAtom(a)->getIdx();
+      neighbors.push_back(o);
+      xyzs.push_back(conf.getAtomPos(o));
+    }
+
+    if (neighbors.size() != 4u) {
+      continue;
+    }
+
+    /* atomParity = sign(det([[1,  1,  1,  1],
+                              [x0, x1, x2, x3],
+                              [y0, y1, y2, y3],
+                              [z0, z1, z2, z3]]))
+     */
+    const auto det = determinant(xyzs[1u].x, xyzs[2u].x, xyzs[3u].x,  //
+                                 xyzs[1u].y, xyzs[2u].y, xyzs[3u].y,  //
+                                 xyzs[1u].z, xyzs[2u].z, xyzs[3u].z) -
+                     determinant(xyzs[0u].x, xyzs[2u].x, xyzs[3u].x,  //
+                                 xyzs[0u].y, xyzs[2u].y, xyzs[3u].y,  //
+                                 xyzs[0u].z, xyzs[2u].z, xyzs[3u].z) +
+                     determinant(xyzs[0u].x, xyzs[1u].x, xyzs[3u].x,  //
+                                 xyzs[0u].y, xyzs[1u].y, xyzs[3u].y,  //
+                                 xyzs[0u].z, xyzs[1u].z, xyzs[3u].z) -
+                     determinant(xyzs[0u].x, xyzs[1u].x, xyzs[2u].x,  //
+                                 xyzs[0u].y, xyzs[1u].y, xyzs[2u].y,  //
+                                 xyzs[0u].z, xyzs[1u].z, xyzs[2u].z);
+    if (det == 0.0) {
+      continue;
+    }
+
+    auto& atomParity = atom.add("cml:atomParity", det > 0.0 ? 1 : -1);
```

Comment:
There is a simpler solution to this problem. I will do a PR against your branch with a suggestion.